clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/KafkaInputRecordReader.java

```diff
@@ -125,147 +85,118 @@ public KafkaInputRecordReader(
       JobConf job,
       DataWriterTaskTracker taskTracker,
       PubSubConsumerAdapter consumer) {
-    this.ownedConsumer = false;
+
     if (!(split instanceof KafkaInputSplit)) {
       throw new VeniceException("InputSplit for RecordReader is not valid split type.");
     }
     KafkaInputSplit inputSplit = (KafkaInputSplit) split;
-    this.consumer = consumer;
-    PubSubPartitionSplit pubSubPartitionSplit = inputSplit.getSplit();
-    this.topicPartition = pubSubPartitionSplit.getPubSubTopicPartition();
-    this.startingPosition = pubSubPartitionSplit.getStartPubSubPosition();
-    this.currentPosition = pubSubPartitionSplit.getStartPubSubPosition();
-    this.endingPosition = pubSubPartitionSplit.getEndPubSubPosition();
-    this.totalRecordsToRead = pubSubPartitionSplit.getNumberOfRecords();
-    this.splitIndex = pubSubPartitionSplit.getSplitIndex();
-    this.splitStartingRecordIndex = pubSubPartitionSplit.getStartIndex();
-    this.shouldUseLocallyBuiltIndexAsOffset = !job.getBoolean(
-        PUBSUB_INPUT_SECONDARY_COMPARATOR_USE_NUMERIC_RECORD_OFFSET,
-        DEFAULT_PUBSUB_INPUT_SECONDARY_COMPARATOR_USE_NUMERIC_RECORD_OFFSET);
+
     this.isSourceVersionChunkingEnabled = job.getBoolean(KAFKA_INPUT_SOURCE_TOPIC_CHUNKING_ENABLED, false);
     String keySchemaString = job.get(KAFKA_SOURCE_KEY_SCHEMA_STRING_PROP);
     if (keySchemaString == null) {
       throw new VeniceException("Expect a value for the config property: " + KAFKA_SOURCE_KEY_SCHEMA_STRING_PROP);
     }
     this.keySchema = Schema.parse(keySchemaString);
-    // In case the Kafka Consumer is passed by the caller, we will unsubscribe all the existing subscriptions first
-    if (!ownedConsumer) {
-      this.consumer.batchUnsubscribe(this.consumer.getAssignment());
-    }
-    this.consumer.subscribe(topicPartition, currentPosition, true);
     this.taskTracker = taskTracker;
+    // Consumer ownership: if caller provides consumer, we don't own it
+    this.ownedConsumer = (consumer == null);
+    consumer = (consumer != null) ? consumer : createConsumer(job);
+    PubSubPartitionSplit pubSubSplit = inputSplit.getSplit();
+    this.topicPartition = pubSubSplit.getPubSubTopicPartition();
+    boolean useLogicalIndexOffset = !job.getBoolean(
```

Comment:
can we use `boolean useNumericRecordOffset = job.getBoolean(PUBSUB_INPUT_SECONDARY_COMPARATOR_USE_NUMERIC_RECORD_OFFSET,
        DEFAULT_PUBSUB_INPUT_SECONDARY_COMPARATOR_USE_NUMERIC_RECORD_OFFSET);` ?